ppsci/arch/chemprop_molecule_utils.py

```diff
@@ -0,0 +1,2425 @@
+            )
+        self.num_lrs = len(optimizer._param_groups)
+        self.optimizer = optimizer
+        self.warmup_epochs = np.array(warmup_epochs)
+        self.total_epochs = np.array(total_epochs)
+        self.steps_per_epoch = steps_per_epoch
+        self.init_lr = np.array(init_lr)
+        self.max_lr = np.array(max_lr)
+        self.final_lr = np.array(final_lr)
+        self.current_step = 0
+        self.lr = init_lr
+        self.warmup_steps = (self.warmup_epochs * self.steps_per_epoch).astype(int)
+        self.total_steps = self.total_epochs * self.steps_per_epoch
+        self.linear_increment = (self.max_lr - self.init_lr) / self.warmup_steps
+        self.exponential_gamma = (self.final_lr / self.max_lr) ** (
+            1 / (self.total_steps - self.warmup_steps)
+        )
+        super(NoamLR, self).__init__(optimizer.get_lr())
+
+    def get_lr(self) -> List[float]:
+        """
+        Gets a list of the current learning rates.
+
+        :return: A list of the current learning rates.
+        """
+        return list(self.lr)
+
+    def step(self, current_step: int = None):
+        """
+        Updates the learning rate by taking a step.
+
+        :param current_step: Optionally specify what step to set the learning rate to.
+                             If None, :code:`current_step = self.current_step + 1`.
+        """
+        if current_step is not None:
+            self.current_step = current_step
+        else:
+            self.current_step += 1
+        for i in range(self.num_lrs):
+            if self.current_step <= self.warmup_steps[i]:
+                self.lr[i] = (
+                    self.init_lr[i] + self.current_step * self.linear_increment[i]
+                )
+            elif self.current_step <= self.total_steps[i]:
+                self.lr[i] = self.max_lr[i] * self.exponential_gamma[i] ** (
+                    self.current_step - self.warmup_steps[i]
+                )
+            else:
+                self.lr[i] = self.final_lr[i]
+            self.optimizer._param_groups[i]["learning_rate"] = self.lr[i]
+
+
+def activate_dropout(module: paddle.nn.Layer, dropout_prob: float):
+    """
+    Set p of dropout layers and set to train mode during inference for uncertainty estimation.
+
+    :param model: A :class:`~chemprop.models.model.MoleculeModel`.
+    :param dropout_prob: A float on (0,1) indicating the dropout probability.
+    """
+    if isinstance(module, paddle.nn.Dropout):
+        module.p = dropout_prob
+        module.train()
+
+
+# === nn_util end   ===
+
+# === features_generators start ===
+Molecule = Union[str, Chem.Mol]
+FeaturesGenerator = Callable[[Molecule], np.ndarray]
```

Comment:
因为你这里进行了实际的类型赋值操作，所以annotations是没用的，给类加上引号可以解决这个问题
```suggestion
Molecule = Union[str, "Chem.Mol"]
FeaturesGenerator = Callable[[Molecule], np.ndarray]
```

Reply:
好的好的